### ISSUE TITLE
rules var-naming[no-role-prefix] and var-naming[pattern]  collide if var_naming_pattern is changed from default

### ISSUE THREAD

**Issue by mhallin2** [ansible/ansible-lint#4142]
##### Summary
Role variable prefix does not work if var_naming_pattern: has been altered from its default.
#Bradley Thornton as requested me to create this as a bug.

If the var_naming_pattern setting in .ansible-lint.cfg file is changed from its default
and you create a role and use set_fact to set a variable 
The rules var-naming[no-role-prefix] and var-naming[pattern]  will collide as both scenarios can not be fulfilled
##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 24.2.3 using ansible-core:2.16.6 ansible-compat:4.1.11 ruamel-yaml:0.18.6 ruamel-yaml-clib:0.2.8
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

update .ansible-lint with the below setting
```.ansible-lint.cfg (paste below)
var_naming_pattern: "^[A-Z_][A-Za-z0-9_]*$"
```
Add a variable to the role vars main.yml file

```main.yml (paste below)
---
Register_DNS_Host_Record_Variable: dummyvalue

```

```console (paste below)
ansible-lint /collections/hosting/network/roles/register_dns_host_record
```

##### Desired Behavior

No errors should be reported
##### Actual Behavior
var-naming[no-role-prefix]: Variables names from within roles should use register_dns_host_record_ as a prefix. (vars: Register_DNS_Host_Record_Variable)

**Comment by Qalthos** [comment]
The `var-naming[no-role-prefix]` rule checks that role variables start with the role name _exactly_, i.e. including case. This is intentional to ensure consistency and clarity of role variables.

I can see in this case why you might want to use different casing, even if the prefix is otherwise the same, but we do not presently support this and are not planning on working on it at present. However, we do always welcome community contributions, and if you put together a PR, we will absolutely consider it.

**Comment by cidrblock** [comment]
Let's talk about this one. I think we can expose the regex to the user.

**Comment by mhallin2** [comment]
@cidrblock any update on this one?

**Comment by mhallin2** [comment]
@Qalthos it is not that I want to change the Role prefix as such.
But the 2 roles collides.  

var_naming_pattern is set for require variable name to start with uppercase 
Since this var-naming[pattern] also applies to roles the role variable needs to be Myrole_Value
But the var-naming[no-role-prefix] rule hard codes it to it needs to be myrole_Value

So one of them will always fail. 
This results in that one of the rules has to be disabled to have the linting to pass.
This is not an issue if you do not change the var_naming_pattern but once done this issue occurs.

What I'm asking for is to have the possibility to configure ansible lint to support this scenario. 
I fully support that the prefix should be there but I don't see why ansible-lint does not support var_naming_pattern all the way into roles.
